### DISCUSSION TITLE
Texas Instruments CC2652P / CC1352P +20 dBm Zigbee 3.0 SoC based SMD modules now available as Ebyte E72-2G4M20S1E, Ebyte E79-400DM2005S, and RF-star RF-BM-2652P2

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#638] [General]
TI CC2652P and CC1352P multi-protocol +20 dBm SoCs based SMD modules now available as Ebyte E72-2G4M20S1E and E79-400DM2005S

Ebyte E72-2G4M20S1E (CC2652P SMD module without IPEX-connector/uFL-connector)

https://www.ebyte.com/en/product-view-news.aspx?id=1002
https://www.ebyte.com/en/downpdf.aspx?id=1002

![image](https://user-images.githubusercontent.com/6320001/105473370-558f2f00-5c9d-11eb-9fa4-b4e732acafd4.png)

Ebyte E79-400DM2005S (CC1352P SMD module with IPEX-connector/uFL-connector)

https://www.ebyte.com/en/product-view-news.aspx?id=766
https://www.ebyte.com/en/downpdf.aspx?id=766

![image](https://user-images.githubusercontent.com/6320001/105476874-6f327580-5ca1-11eb-8b36-e6d5129179ff.png)

So expect that several CC2652P and CC1352P based USB dongles/sticks/adapters, shields, and bridges/gateways will soon be available.

Ebyte E72-2G4M20S1E (CC2652P SMD module) on Aliexpress
https://www.aliexpress.com/item/1005001265532720.html
https://www.aliexpress.com/item/1005001315624806.html
https://www.aliexpress.com/item/1005001825198001.html
https://www.aliexpress.com/item/1005001862475194.html
https://www.aliexpress.com/item/1005001297695466.html
https://www.aliexpress.com/item/4001297468233.html
https://www.aliexpress.com/item/1005001481847788.html
https://www.aliexpress.com/item/1005001647897822.html

Ebyte E79-400DM2005S (CC1352P SMD module) on Aliexpress
https://www.aliexpress.com/item/4000321518715.html
https://www.aliexpress.com/item/4000321960037.html
https://www.aliexpress.com/item/1005001456367381.html
https://www.aliexpress.com/item/1005001829722791.html
https://www.aliexpress.com/item/4001339890172.html
https://www.aliexpress.com/item/1005001615475580.html
https://www.aliexpress.com/item/1005001483622276.html

Personally, I am interested in these RF SoC SMD modules for Zigbee via zigpy as the "P" in name standard for their built-in Power Amplifier with +20 dBm amplification, but the SoC + SDKs also supports Thread, Bluetooth 5 Low Energy, and IEEE 802.15.4g proprietary protocols.

https://www.ti.com/product/CC2652P
https://www.ti.com/lit/gpn/cc2652p

### Comparsion

Summarized info posted on Electrolama project zig-a-zig-ah (a.k.a. zzh) for which compare similar chips from Texas Instruments.

https://electrolama.com/projects/zig-a-zig-ah/

#### The newer generation:

- CC2652P = CC2652R with a built-in RF PA. Not pin or firmware compatible with CC2652R/CC2652RB. The design for zzh-p uses this CC2652P SoC.
- CC2652R = 2.4GHz only multi-protocol (Zigbee, Bluetooth, Thread, ...) wireless MCU. Cortex-M0 core for radio stack and Cortex-M4F core for application use, plenty of RAM. Free compiler option from TI. zzh uses this chip.
-  CC2652RB = Pin compatible "Crystal-less" CC2652R (so you could use it if you were to build your own zzh and omit the crystal) but not firmware compatible.

#### The "middle" generation:

-  CC2538: 2.4GHz Zigbee, 6LoWPAN, and IEEE 802.15.4 wireless MCU. ARM Cortex-M3 core with with 512kB Flash and 32kB RAM.

#### The older generation:

- CC2530 = 2.4GHz Zigbee and IEEE 802.15.4 wireless MCU. Intel 8051 core, 256 Flash, only has 8kB RAM. Needs expensive compiler license for official TI stack to built your own firmware.
- CC2531 = CC2530 with built-in UART/TTL to USB bridge. Used in the cheap "Zigbee sticks" sold everywhere. Intel 8051 core, 256 Flash, only has 8kB RAM. Needs expensive compiler license for official TI stack to built your own firmware.

**Comment by Hedda** [comment]
https://electrolama.com/projects/zig-a-zig-ah/#Flavours

"**_zzh-p is the experimental version with CC2652P (built-in PA). Shares most of the same design with zzh except for the RF parts and will fit in a slightly larger case. It is a work in progress, sign up to the Electrolama mailing list to be notified of project updates and when kits/assembled units go on sale. A sneak peek:_**"

![image](https://user-images.githubusercontent.com/6320001/104472617-6ca69080-55bc-11eb-9d7c-a6221d27391b.png)

**Comment by Hedda** [comment]
RF-BM-2652P2 from RF-star / RFSTAR is an alternative CC2652P based SMD module

https://www.szrfstar.com/product/237-en.html

https://www.szrfstar.com/download/604-en.html

https://www.aliexpress.com/item/1005001677721370.html

![image](https://user-images.githubusercontent.com/6320001/105469111-0d214280-5c98-11eb-9831-c388ae79f85f.png)

![image](https://user-images.githubusercontent.com/6320001/105472324-16aca980-5c9c-11eb-9590-d04f68fc6130.png)

**Reply by MattWestb** [reply]
@Hedda Do you reconicing the pads layout for the external antenna from some cooked ESP32 ETH boards that "some" user have posted before ??

**Reply by Hedda** [reply]
Maybe @tube0013 knows ;). By the way, checkout https://github.com/mercenaruss/zigbee-stick-v4  and https://github.com/Koenkk/Z-Stack-firmware/issues/210

**Reply by tube0013** [reply]
I just got these in.. but my board design was off so waiting on new rev of board.. with a stencil... thanks for the nudge toward reflow ovens. put a diy one together yesterday.

**Reply by Hedda** [reply]
> thanks for the nudge toward reflow ovens. put a diy one together yesterday.

Nice! Yeah, I read that there are also hobby communities around DIY upgrades/mods for popular low-cost reflow ovens from China.

See example https://hackaday.io/project/175048-t-962a-reflow-oven-modifications

**Comment by Hedda** [comment]
Noticed that Koenkk is hosting Z-Stack 3 firmware for several CC2652P and CC1352P adapters tested with zigbee-herdsman/Zigbee2MQTT

https://github.com/Koenkk/Z-Stack-firmware/tree/develop/coordinator/Z-Stack_3.x.0/bin

https://github.com/Koenkk/Z-Stack-firmware/blob/develop/coordinator/Z-Stack_3.x.0/bin/README.md

# Tested adapters

<table>
<thead>
  <tr>
    <th>Adapter</th>
    <th>TI Chip/Module Used</th>
    <th>Firmware to Flash</th>
    <th>BSL Trigger Pin (1)</th>
    <th>Auto-BSL (2)</th>
    <th>RF Switch Control Pins (3)</th>
    <th>LED(s)</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Electrolama zzhp-lite</td>
    <td>CC2652P<br>(Ebyte E72)<br></td>
    <td>CC1352P2_CC2652P_other_*.zip</td>
    <td>DIO_15</td>
    <td>Yes</td>
    <td>DIO_5: 20dBm PA ??<br>DIO_6: 2.4GHz ??<br></td>
    <td>DIO_7 (Pink)</td>
  </tr>
  <tr>
    <td>Electrolama zzhp</td>
    <td>CC2652P</td>
    <td>CC1352P2_CC2652P_other_*.zip</td>
    <td>DIO_15</td>
    <td>Yes</td>
    <td>DIO_5: 20dBm PA ??<br>DIO_6: 2.4GHz ??</td>
    <td>DIO_7 (Pink)</td>
  </tr>
  <tr>
    <td>Electrolama zoe2</td>
    <td>CC1352P<br>(Ebyte E79)<br></td>
    <td>CC1352P2_CC2652P_other_*.zip</td>
    <td>DIO_15</td>
    <td>No</td>
    <td>DIO_5: 20dBm PA ??<br>DIO_6: 2.4GHz ??</td>
    <td>DIO_7 (Pink)</td>
  </tr>
  <tr>
    <td>ZigStar<br></td>
    <td>CC2652P<br>(RFSTAR RF-BM-2652P2)<br></td>
    <td>CC1352P2_CC2652P_launchpad_*.zip</td>
    <td>DIO_15</td>
    <td>Only for CH340C ver.</td>
    <td>DIO_28: 2.4Ghz<br>DIO_29: 20dBm PA</td>
    <td>DIO_6 (Green)<br>DIO_7 (Red)<br></td>
  </tr>
  <tr>
    <td>Zigbeer E72 by Egony</td>
    <td>CC2652P<br>(Ebyte E72)</td>
    <td>CC1352P2_CC2652P_other_*.zip</td>
    <td>DIO_15</td>
    <td>No</td>
    <td>DIO_5: 20dBm PA<br>DIO_6: 2.4GHz</td>
    <td>DIO_8 (Green)<br>DIO_7 (Red)<br></td>
  </tr>
  <tr>
    <td>Zigbeer E79 by Egony</td>
    <td>CC1352P<br>(Ebyte E79)</td>
    <td>CC1352P2_CC2652P_other_*.zip</td>
    <td>DIO_15</td>
    <td>No</td>
    <td>DIO_5: 20dBm PA ?<br>DIO_6: 2.4GHz ?</td>
    <td>DIO_8 (Green) ?<br>DIO_7 (Red) ?<br></td>
  </tr>
</tbody>
</table>

**Comment by Hedda** [comment]
Several CC2652P and CC1352P based USB dongles/sticks/adapters and shields are already emerging as open-source hardware designs.

ZigStar stick/dongle project by @mercenaruss is based the above mentioned RF-Star RF-BM-2652P2 SMD-module
* https://mercenaruss.github.io/zigbee-stick-v4/
  * https://github.com/mercenaruss/zigbee-stick-v4

Zigbeer E72 and Zigbeer E79 sticks/dongles and shield projects by @egony are based on the above mentioned Ebyte SMD-modules
* https://github.com/egony/cc2652p_E72-2G4M20S1E/wiki
  * https://github.com/egony/cc2652p_E72-2G4M20S1E

Electrolama zzhp / zzh-p and zoe2 designs by @electrolama / @omerk are based on SMC CC2652P/CC1352P SoCs directly from TI
* https://github.com/electrolama/
  * https://electrolama.com/projects/zig-a-zig-ah/
  * https://electrolama.com/projects/zoe/

PS: I also suspect that @slaesh is or will also make a CC2652P based dongle as an update to his popular CC2652R/CC2652R based stick.
* https://slae.sh/projects/cc2652/
  * https://github.com/slaesh/cc2652-stick/

**Comment by Hedda** [comment]
By the way, there relative discussion in https://github.com/Koenkk/Z-Stack-firmware/issues/210 for hardware developers which discusses community standardisation of BSL and RF Switch pins and an LED pin(s) for all Texas Instruments CC2652 and CC1352 based open-source hardware board designs.
